utils.py

```diff
@@ -1,69 +1,22 @@
     
-    Args:
-        df (object): can be a variety of dataframe like objects
-        
-    Returns:
-        pandas.DataFrame: a dataframe that is ready to be serialized
-    """
-    import pandas as pd
-    import numpy as np
-
-    df = df.copy()
-
-    for col_name, dtype in df.dtypes.iteritems():
-        if str(dtype) == 'category':
-            # XXXX: work around bug in to_json for categorical types
-            # https://github.com/pydata/pandas/issues/10778
-            df[col_name] = df[col_name].astype(str)
-        elif np.issubdtype(dtype, np.integer):
-            # convert integers to objects; np.int is not JSON serializable
-            df[col_name] = df[col_name].astype(object)
-        elif np.issubdtype(dtype, np.floating):
-            # For floats, convert nan->None: np.float is not JSON serializable
-            col = df[col_name].astype(object)
-            df[col_name] = col.where(col.notnull(), None)
-        elif str(dtype).startswith('datetime'):
-            # Convert datetimes to strings
-            # astype(str) will choose the appropriate resolution
-            df[col_name] = df[col_name].astype(str).replace('NaT', '')
-    return df
-
-def fill_spec_with_data(spec, data):
-    """Take a Vega specification with missing data elements and complete it
-    * Uses an incomplete Vega specification that has named data arguments
-        which have empty values and replaces those missing values with
-        data elements passed in
-    * The Vega specification needs to have named data arguments that match
-        the data that is passed in to this function
-        
-    Args:
-        spec (dictionary): the Vega specification with incomplete data
-        data (named dictionary of pandas.DataFrame): data elements
-        
-    Returns:
-        dictionary: a completed Vega specification ready to be used
-    """
-    data_names = [d['name'] for d in spec['data']]
-    
-    for name, df in data.items():
-        if name not in data_names:
-            raise ValueError('Name not in data spec')
-        for ix in range(len(data_names)):
-            if data_names[ix] == name:
-                break
-                
-        df = sanitize_dataframe(df)
-        spec['data'][ix]['values'] = df.to_dict(orient='records')
-
-    return spec
+    return (gg.theme_bw(base_size = 14 * fontsize_mult) +
+        gg.theme(
+          line = gg.element_line(color = "#4d4d4d"), 
+          rect = gg.element_rect(fill = "white", color = None), 
+          text = gg.element_text(color = "black"), 
+          axis_ticks = gg.element_line(color = "#4d4d4d"),
+          legend_key = gg.element_rect(color = None), 
+          panel_border = gg.element_rect(color = "#4d4d4d"),  
+          panel_grid = gg.element_line(color = "#b3b3b3"), 
+          panel_grid_major_x = gg.element_blank(),
+          panel_grid_minor = gg.element_blank(),
+          strip_background = gg.element_rect(fill = "#e5e5e5", color = None),
```

Comment:
Consider changing `#e5e5e5` to `#FEF2E2` for `strip.background` color to make the facet strips a little less drab.

Reply:
Will do